packages/quantic/force-app/main/default/lwc/quanticCaseClassification/quanticCaseClassification.js

```diff
@@ -202,6 +202,9 @@ export default class QuanticCaseClassification extends LightningElement {
           this.showSelect();
         }
         this.setFieldValue(value, true);
+      } else {
+        const value = this.field.state.value;
```

Comment:
The fix itself seems correct, however I don't really understand why hitting the `back` button ends up executing this `else` block.

It seems many small fixes were made in `updateFieldState` and the function could now use some refactoring. My understanding is that, most of that logic depends on whether the field has already been selected. So maybe we should detect this condition first, and then do the rest of the logic.